src/MarginTrading.DataReader/Controllers/RoutesController.cs

```diff
@@ -23,30 +27,44 @@ public RoutesController(IMatchingEngineRoutesRepository routesRepository)
         /// </summary>
         [HttpGet]
         [Route("")]
-        public async Task<IEnumerable<MatchingEngineRoute>> GetAllRoutes()
+        public async Task<List<MatchingEngineRouteContract>> List()
         {
-            return (await _routesRepository.GetAllRoutesAsync()).Select(TransformRoute);
+            return (await _routesRepository.GetAllRoutesAsync()).Select(TransformRoute).ToList();
         }
 
         /// <summary>
         /// Gets a route by <paramref name="id"/>
         /// </summary>
         [HttpGet]
         [Route("{id}")]
-        public async Task<MatchingEngineRoute> GetRouteById(string id)
+        public async Task<MatchingEngineRouteContract> GetById(string id)
         {
             return TransformRoute(await _routesRepository.GetRouteByIdAsync(id));
         }
 
-        private static MatchingEngineRoute TransformRoute(IMatchingEngineRoute route)
+        private static MatchingEngineRouteContract TransformRoute(IMatchingEngineRoute route)
         {
             string GetEmptyIfAny(string value)
             {
                 const string AnyValue = "*";
                 return value == AnyValue ? null : value;
             }
+            OrderDirectionContract? GetDirection(OrderDirection? direction)
+            {
+                switch (direction)
+                {
+                    case null:
+                        return null;
+                    case OrderDirection.Buy:
+                        return OrderDirectionContract.Buy;
+                    case OrderDirection.Sell:
+                        return OrderDirectionContract.Sell;
+                    default:
+                        throw new System.Exception("Invalid OrderType");
```

Comment:
It's better to throw ArgumentException, passing the direction var name and value

Reply:
We have EnumExtensions.ToType for such transformations.